File: 383-ransom-note/383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        for each in ransomNote:
            x = magazine.find(each)
            if x>-1:
                magazine = magazine[:x] + magazine[x+1:]
            else:
                return False
            
        return True

File: 383-ransom-note/383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        map<char,int>mr, mm;
        for(int i = 0;i<ransomNote.size(); i++){
            mr[ransomNote[i]]++;
        }
        for(int i = 0; i<magazine.size(); i++){
            mm[magazine[i]]++;
        }
        for(int i = 0; i<ransomNote.size(); i++){
            if(mr[ransomNote[i]]>mm[ransomNote[i]]) 
                return false;
        }
        return true;
    }
};